src/internal/hooks/use-base-component/styles-check.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { useEffect } from 'react';
+
+import { GIT_SHA, PACKAGE_VERSION, THEME } from '../../environment';
+import { metrics } from '../../metrics';
+
+export function checkMissingStyles() {
+  const result = getComputedStyle(document.body).getPropertyValue(`--awsui-version-info-${GIT_SHA}`);
+  if (!result) {
+    console.error(
+      `Missing Cloudscape CSS for theme "${THEME}", version "${PACKAGE_VERSION}", and git sha "${GIT_SHA}".`
+    );
+    metrics.sendOpsMetricObject('awsui-missing-css-asset', {});
+  }
+}
+
+export function idleWithDelay(cb: () => void) {
```

Comment:
Exported `checkMissingStyles` and `idleWithDelay` as separate functions to allow write simpler unit tests on each part of the functionality